packages/server/lib/plugins/util.coffee

```diff
@@ -32,7 +32,7 @@ module.exports = {
 
   wrapChildPromise: (ipc, invoke, ids, args = []) ->
     Promise.try ->
-      return invoke(ids.eventId, args)
+      return invoke(ids.eventId, args, ipc)
```

Comment:
`ipc` was not being passed as a parameter to `invoke`. 

```javascript
sendError(new Error(`No handler registered for event id ${eventId}`))
```

The code abobe, was throwing an exception because of this, an there was no test for this. I can create tests to both validations, but before i want to know if this is a good solution.

Thanks!

Reply:
Instead of catching this here, I think it would be better to validate the user's input when they call `on('file:preprocessor'). That validation would go [here](https://github.com/cypress-io/cypress/blob/develop/packages/server/lib/plugins/child/run_plugins.js#L37). Then we'll have a guarantee that a handler is set when this code runs.

Since different events can require different types of handlers, I'd recommend utilizing a map like the following:

```javascript
const eventValidators = {
  'file:preprocessor': _.isFunction,
  'before:browser:launch': _.isFunction,
  'task': _.isPlainObject,
  'after:screenshot': _.isFunction,
}
```

While we're at it, let's go ahead and validate that the event name is being passed in (since they could call `on()` with no arguments) and validate that it's a valid event name. That could help someone in case they made a typo too.